collections/item.go

```diff
@@ -0,0 +1,48 @@
+package collections
+
+import (
+	"fmt"
+
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+)
+
+// NewItem instantiates a new Item instance, given the value encoder of the item V.
+func NewItem[V any](sk storetypes.StoreKey, prefix Prefix, valueEncoder ValueEncoder[V]) Item[V] {
+	return (Item[V])(NewMap[noKey, V](sk, prefix, noKey{}, valueEncoder))
+}
```

Comment:
This code says an Item (e.g. 123) is expressed within a given StorageProvider as a constant key (no_key) with a variable value (123), instead of e.g. a variable key (value:123) with a constant value (<nothing>). But that means that a given StorageProvider can only ever contain a single Item. Is that the intent?

Reply:
Yes. Item should only contain one single instance of a specific object. 

Example: Parameters of a module, you will only have one Parameter instance in a module. A sequence (nonce) you will have one ever increasing nonce item. etc.

Reply:
> one single instance of a specific object

Then this is not really an Item, I don't think. Item implies identity. But this code says that Item is a single thing regardless of identifier.

Reply:
@peterbourgon, as mentioned above, I am not attached to nomenclature, rationale for using this name was to bring continuity between cosmwasm contract devs and cosmos-sdk module devs.

cw-storage-plus defines an `Item` API whose equivalent is collections's `Item` API, reference: https://github.com/CosmWasm/cw-storage-plus/blob/HEAD/src/item.rs

Reply:
That Item defines its storage_key based on user input. Your Item doesn't do that.

Reply:
it does though, through the `Prefix`?

Reply:
Prefix is a namespace, not an identifier, isn't it?

Reply:
Well, yes and no, prefixes in the SDK are used as namespaces and identifiers. This happens in cosmwasm too, eg:

```
| type | prefix  | key | value |
| map | "accounts" | AccAddress | Account |
| item | "params" | none | Params |
| item | "nonce" | none | uint64
```

This is not ideal, but it allows us to retain backwards compatibility with what was done so far.

Reply:
resolving, if it's not good enough then feel free to unresolve